addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -40,19 +40,20 @@ def _shorten_links(self, html, link_tracker_vals, blacklist=None, base_url=None)
             long_url = match[1]
             label = (match[3] or '').strip()
 
-            if not blacklist or not [s for s in blacklist if s in long_url] and not long_url.startswith(short_schema):
+            if not (long_url.startswith(short_schema) or (blacklist and [s for s in blacklist if s in long_url])):
```

Comment:
Assuming these changes wer implemented for performance reasons, the list comprehension should probably be something like `any(s in long_url for s in blacklist)` to avoid collecting an actual list completely unnecessarily, and limit the iteration on match.